ui/src/services/logs-tracker-service/logs-tracker-service.ts

```diff
@@ -0,0 +1,101 @@
+import { injectable } from 'inversify'
+import { makeAutoObservable } from 'mobx'
+
+import { socket } from '@/api/socket'
+import { LogcatEntryMessage } from '@/types/logcat-entry-message.type'
+
+import { debounce } from '@/lib/utils/debounce.util'
+
+import type { DeviceLogs } from './types'
+
+@injectable()
+export class LogsTrackerService {
+  private batchLogsSize = 25
+
+  /* NOTE: Do not mutate this object to ensure stable references for React Table.
+    For more details, see: https://tanstack.com/table/latest/docs/guide/data#give-data-a-stable-reference
+  */
+  logsByDeviceSerial: Record<string, DeviceLogs> = {}
+  batchedLogs: LogcatEntryMessage[] = []
+  maxLogsBuffer = 3000
+  maxVisibleLogs = 100
+
+  constructor() {
+    makeAutoObservable(this, {
+      batchedLogs: false,
+    })
+
+    this.onLogcatEntry = this.onLogcatEntry.bind(this)
+    this.debouncedFlushLogs = debounce(this.flushLogs.bind(this), 100)
+
+    socket.on('logcat.entry', this.onLogcatEntry)
+  }
+
+  initializeLogcat(serial: string): void {
+    if (this.logsByDeviceSerial[serial]) {
+      this.setLogcatStarted(serial, true)
+    }
+
+    if (!this.logsByDeviceSerial[serial]) {
+      this.logsByDeviceSerial = {
+        ...this.logsByDeviceSerial,
+        [serial]: {
+          logs: [],
+          isLogcatStarted: true,
+        },
+      }
+    }
+  }
+
+  turnOffLogcat(serial: string): void {
+    this.setLogcatStarted(serial, false)
+  }
+
+  clearDeviceLogs(serial: string): void {
```

Comment:
This should also be called when we release the device or the device panel is no longer shown/used